File: TomadaObjeto/Teste.java
package TomadaObjeto;

public class Teste {
    public static void main(String[] args) {
        Tomada3Pinos tomada3Pinos = new Tomada3Pinos();
        Tomada2Pinos tomada2Pinos = new TomadaAdapter(tomada3Pinos);
        tomada2Pinos.ligarNaTomada2Pinos();
    }
}

File: TomadaClasse/Teste.cpp
#include "Tomada2Pinos.h"
#include "TomadaAdapter.h"

int main() {
    Tomada2Pinos* tomada2Pinos = new TomadaAdapter();
    tomada2Pinos->ligarNaTomada2Pinos();
    delete tomada2Pinos;
    return 0;
}